torch/jit/_isinstance.py

```diff
@@ -0,0 +1,107 @@
+from typing import List, Dict, Tuple, Union, Optional
+
+
+def get_origin(target_type):
+    return getattr(target_type, "__origin__", None)
+
+
+def get_args(target_type):
+    return getattr(target_type, "__args__", None)
```

Comment:
I think the code works here because in 3.6 `List` is treated differently from `List[str]` : 

```
Python 3.6.2 |Anaconda, Inc.| (default, Oct  5 2017, 07:59:26) 
[GCC 7.2.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from typing import List
>>> a = List[str]
>>> a.__args__
(<class 'str'>,)
>>> a.__origin__
typing.List
>>> a = List
>>> a.__args__
>>> a.__origin__
```

But this allows eager mode to accept List/Dict/Tuple as the object type while Scripting doesn't for python 3.6. I'll see what I can do to make the behavior the same